src/hazelcore/semantics/dynamics/Dynamics.re

```diff
@@ -8,6 +8,29 @@ module Pat = {
     | Expands(DHPat.t, HTyp.t, Contexts.t, Delta.t)
```

Comment:
I moved it to its own module. I considered writing a `map` function when I was writing this too, but this seemed more future proof, since we could use any function that works on option this way. Otherwise we keep duplicating `Option` functions in this module.
What do you think about this?

Reply:
Thanks. Adding `Option`-like functions as needed sounds fine to me. Either there's a cost of adding those functions, or there's the cost of converting into `Option` types and back whenever we wanna piggyback on the `Option` type functions, and I'd rather pay the former.

Reply:
Okay, now I looked at it more closely, the `Option` type is also required by the use of `map_with_accumulator_opt`. I could make a similar function `map_with_accumulator_expand_result`, but then we're also duplicating `ListUtil` functions, not just `Option`. Should I still go through with it?